ddl/partition.go

```diff
@@ -884,6 +899,16 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 			return ver, errors.Trace(err)
 		}
 		physicalTableIDs = removePartitionInfo(tblInfo, partNames)
+	}
+
+	rules := buildPlacementDropRules(job.SchemaID, tblInfo.ID, physicalTableIDs)
+
+	err = infosync.UpdatePlacementRules(nil, rules)
+	if err != nil {
+		return ver, errors.Wrapf(err, "failed to notify PD the placement rules")
+	}
+
+	if job.Type != model.ActionAddTablePartition {
 		ver, err = updateVersionAndTableInfo(t, job, tblInfo, true)
```

Comment:
Could we put it after line921? If `job.Type != model.ActionAddTablePartition` is true, it seams that the job isn't rollingback.

Reply:
I do not quite understand it. This is how it is originally.

It will first try to `updateVersionAndTableInfo` if `job.Type != model.ActionAddTablePartition`. Then it is the logic of rollingback. Is this a bug of the original code?

Reply:
This is not to say that the original logic is wrong. We can understand that `job.type = = model.actionadtablepartition' and job.IsRollingback ()' are similar conditions.

Reply:
So you mean `==ActionAddTablePartition` is synonymous to `isRollingback`. Because `addTable` is the only one action that converts to `dropPartition` for rollback.

And I can put it in the if-else of `isRollingback` to make intent of the code more clear.

Reply:
I've searched the whole repo using grep. Yes, it is definitely synonymous.

Reply:
> So you mean `==ActionAddTablePartition` is synonymous to `isRollingback`. Because `addTablePartition` is the only one action that converts to `dropPartition` for rollback.
> 
> And I can put it in the if-else of `isRollingback` to make intent of the code more clear.
Yes.

But I think the original code is problematic related #19317.